crates/evm/core/src/snapshot.rs

```diff
@@ -39,6 +39,20 @@ impl<T> Snapshots<T> {
         snapshot
     }
 
+    /// Removes all snapshots after `id`.
+    ///
+    /// This will not remove the `id` itself, only clone its data, since the snapshot will still
```

Comment:
this make it impossible to delete snapshots?

same issue as https://github.com/foundry-rs/foundry/issues/6411

Reply:
makes it impossible to delete the very first snapshot. but reverting to the oldest one should revert all other ones. so arguably not a memory leak?

or did I misunderstand your point?
anyway, probably a cheatcode for cleaning snapshots, as you suggested in that issue, makes sense. I can also take a shot at that, but probably as a separate PR?

Reply:
yeah same issue:

think of the vm calls as rpc calls, this will never delete snapshots:
```
for(uint cc = 0; cc < 10000; cc++) {
                preState = vm.snapshot();
                vm.revertTo(preState);
            }
```

a followup PR for this would be great @naps62